### PR TITLE
Hooking up cli args to livereload tunnel

### PR DESCRIPTION
**Pull request by joshuaskelton**
## Summary
This fixes livereload tunnel functionality by correctly setting the tunnel option.

## Changes
- Changed hookLiveReload interface to take an options object.
- hookLiveReload now uses the tunnel option.
- Updated LiveReloadOptions typing.

### CODE REVIEW

**Comment by nisheetjain** [comment]
@omefire can you please review this?

**Comment by omefire** [comment]
sure,

File: src/typings/tacoLiveReload.d.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ declare module TacoLiveReload {
     export interface LiveReloadOptions {
         ignore: string,
         ghostMode: boolean,
+        tunnel: boolean,
```
Comment (omefire) [inline src/typings/tacoLiveReload.d.ts]: Please, add a brief description of the 'tunnel' option

File: src/taco-cli/cli/liveReload.ts [hunk 2]
```diff
@@ -29,15 +29,16 @@ import CordovaHelper = tacoUtility.CordovaHelper;
  */
 class LiveReload {
 
-    private static hookLiveReload(liveReloadEnabled: boolean, deviceSyncEnabled: boolean, platforms: string[], ignore?: string): Q.Promise<any> {
+    private static hookLiveReload(options: any, platforms: string[], ignore?: string): Q.Promise<any> {
```
Comment (omefire) [inline src/taco-cli/cli/liveReload.ts]: Please, make 'ignore' a part of 'options'.

**Comment by omefire** [comment]
Overall, LGTM! 
Though I had some minor comments...

File: src/typings/tacoLiveReload.d.ts (hunk 1 above)
Comment (joshuaskelton) [inline src/typings/tacoLiveReload.d.ts]: Done!

Comment (meenakb) [inline src/typings/tacoLiveReload.d.ts]: Are these options exposed in commands.json? In other words, shouldn't we enumerate these "new" options are supported options for "run" and "emulate" on "taco help"?

Comment (joshuaskelton) [inline src/typings/tacoLiveReload.d.ts]: I've added a description for tunnel. The ghostMode option is the internal name for deviceSync. We should document the ignore option, I'll get going on that.

Comment (joshuaskelton) [inline src/typings/tacoLiveReload.d.ts]: image

Comment (meenakb) [inline src/typings/tacoLiveReload.d.ts]: LGTM

**Comment by nisheetjain** [comment]
:+1:

File: src/typings/tacoLiveReload.d.ts (hunk 1 above)
Comment (omefire) [inline src/typings/tacoLiveReload.d.ts]: The --ignore option is also applicable for devicesync

Comment (joshuaskelton) [inline src/typings/tacoLiveReload.d.ts]: What does it mean for devicesync to ignore changes to specific files?

Comment (omefire) [inline src/typings/tacoLiveReload.d.ts]: --devicesync works exactly like --LiveReload except that in addition to everything else, it syncs user actions across browsers.

Comment (joshuaskelton) [inline src/typings/tacoLiveReload.d.ts]: Shouldn't devicesync then just be an option to livereload?

**Comment by nisheetjain** [comment]
:+1:

**Pull request merged**